modules/stadeniolib/main.nf

```diff
@@ -0,0 +1,41 @@
+process STADENIOLIB {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::staden_io_lib=1.14.14" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/staden_io_lib:1.14.14--h0d9da7e_3' :
+        'quay.io/biocontainers/staden_io_lib:1.14.14--h0d9da7e_3' }"
+
+    input:
+    tuple val(meta), path(bam)
+    path(fasta)
+    path(fai)
+
+    output:
+    tuple val(meta), path("*.cram") ,emit: cram
+    path "versions.yml"             ,emit: versions
```

Comment:
You also need to make an optional output for BAM index files

```
    -g FILE        Convert to Bam using index (file.gzi)
    -G FILE        Output Bam index when bam input(file.gzi)
```

Reply:
I'm not really sure those are relevant (hence why I left them out), since the default output is set to cram. The tool supports converting sam/bam/cram to sam/bam/cram, but since we already have samtools for that I defaulted to bam > cram, since that's it only added value

Reply:
I'm going to be picky and say we don't want to assume what a pipeline devs wants to do, so better to support from the get-go so we don't have to do repeated iterations/updates of the same module (speaking from painful experience :grimacing: )

Reply:
I'll update everything and give you a ping when it's ready!